aiven/client/cli.py

```diff
@@ -124,21 +126,26 @@ def get_current_date() -> datetime:
     from typing import Protocol  # pylint: disable=no-name-in-module
 
     class ClientFactory(Protocol):  # pylint: disable=too-few-public-methods
-        def __call__(self, base_url: str, show_http: bool, request_timeout: Optional[int]):
+        def __call__(self, base_url: str, show_http: bool, request_timeout: Optional[int]) -> client.AivenClient:
             ...
 
 
 else:
-    ClientFactory = Callable[..., client.AivenClient]
+    ClientFactory = Callable[..., client.AivenClient]  # type: ignore
```

Comment:
because `mypy` complains that `Cannot assign to a type` :thinking:

Reply:
I think this is because of the other branch of this `if`. I'd drop the whole `if` here and only keep this line with updated arguments of the `Callable`. I don't see a point to define it as a protocol.

Reply:
I think the protocol is to help specify and annotate the keyword arguments (once we're moving to py3.8+).  Because with `Callable` we cannot specify the name of keyword arguments that we expect. In this case if I try 
```python
Callalble[[str, bool, Optional[int]], AivenClient]
```
Then, mypy will complain about e.g `Unexpected keyword argument "base_url"` when we call `client_factory` in [`self.pre_run`](https://github.com/aiven/aiven-client/blob/371f5ead47aefc97a6df821fc45fe3a50a230f53/aiven/client/cli.py#L4640)

Reply:
Ah, right. I'm wondering if something like the following will work?
```python
class ClientFactory(Callable[..., client.AivenClient]):
    pass
```

Otherwise let's keep `type: ignore`.